src/Contexts/FilterContext.js

```diff
@@ -0,0 +1,152 @@
+  };
+
+  const sortbyprice = (data, sortBy) => {
+    let filteredList = data;
+    if (sortBy) {
+      if (sortBy === "0")
+        return filteredList.filter((item) => (item.price <= 0 ? true : false));
+
+      if (sortBy === "200")
+        return filteredList.filter((item) =>
+          item.price <= 200 ? true : false
+        );
+
+      if (sortBy === "400")
+        return filteredList.filter((item) =>
+          item.price <= 400 ? true : false
+        );
+
+      if (sortBy === "600")
+        return filteredList.filter((item) =>
+          item.price <= 600 ? true : false
+        );
+
+      if (sortBy === "800")
+        return filteredList.filter((item) =>
+          item.price <= 800 ? true : false
+        );
+
+      return filteredList;
+    }
+    return filteredList;
+  };
+
+  const sortbycategory = (data, sortBy) => {
+    let filteredList = data;
+    if (
+      sortBy.showManTshirt ||
+      sortBy.showWomenTshirt ||
+      sortBy.showKidsTshirt ||
+      sortBy.showFamilyTshirt
+    ) {
+      const ManCategory = filteredList.reduce(
+        (pre, curr) =>
+          state.showManTshirt && curr.categoryName === "Man"
+            ? [...pre, curr]
+            : [...pre],
+        []
+      );
+      const WomenCategory = filteredList.reduce(
+        (pre, curr) =>
+          state.showWomenTshirt && curr.categoryName === "Women"
+            ? [...pre, curr]
+            : [...pre],
+        ManCategory
+      );
+      const KidsCategory = filteredList.reduce(
+        (pre, curr) =>
+          state.showKidsTshirt && curr.categoryName === "Kids"
+            ? [...pre, curr]
+            : [...pre],
+        WomenCategory
+      );
+      const FamilyCategory = filteredList.reduce(
+        (pre, curr) =>
+          state.showWomenTshirt && curr.categoryName === "Family"
+            ? [...pre, curr]
+            : [...pre],
+        KidsCategory
+      );
+      return FamilyCategory;
+    }
+    return filteredList;
+  };
+
+  const sortbyrating = (data, sortBy) => {
+    let filteredList = data;
+    if (sortBy) {
+      if (sortBy === "4STAR")
+        return filteredList.filter((item) => (item.Rating >= 4 ? true : false));
+
+      if (sortBy === "3STAR")
+        return filteredList.filter((item) => (item.Rating >= 3 ? true : false));
+
+      if (sortBy === "2STAR")
+        return filteredList.filter((item) => (item.Rating >= 2 ? true : false));
+
+      if (sortBy === "1STAR")
+        return filteredList.filter((item) => (item.Rating >= 1 ? true : false));
+
+      return filteredList;
+    }
+    return filteredList;
+  };
+
+  const sortbyrange = (data, sortBy) => {
+    let filteredList = data;
+    if (sortBy) {
+      if (sortBy === "LOW_TO_HIGH")
+        return filteredList.sort((a, b) => a.price - b.price);
+
+      if (sortBy === "HIGH_TO_LOW")
+        return filteredList.sort((a, b) => b.price - a.price);
+
+      return filteredList;
+    }
+    return filteredList;
+  };
```

Comment:
You can extract these function in a separate file

Reply:
Changed in Code ,Thankyou